packages/element-lib/src/sidetree-v2/protocol/resolve.js

```diff
@@ -1,20 +1,77 @@
 /* eslint-disable arrow-body-style */
-const jsonpatch = require('fast-json-patch');
 const { payloadToHash, verifyOperationSignature } = require('../func');
 
 const create = (state, operation) => ({
   ...operation.decodedOperationPayload,
   id: `did:elem:${payloadToHash(operation.decodedOperationPayload)}`,
 });
 
+const applyPatch = (didDocument, patch) => {
+  if (patch.action === 'add-public-keys') {
+    const publicKeySet = new Set(didDocument.publicKey.map(key => key.id));
+    return patch.publicKeys.reduce((currentState, publicKey) => {
+      if (!publicKeySet.has(publicKey)) {
+        return {
+          ...currentState,
+          publicKey: [
+            ...currentState.publicKey,
+            {
+              ...publicKey,
+              // FIXME: Need controller property to be compliant with the protocol
```

Comment:
nit: create an issue for this / link it in the comment

Reply:
After a closer look, it looks like the controller property is optional and not used in the protocol so I we dont have to worry about it right now. Removing the comment then merging.

https://github.com/decentralized-identity/sidetree/search?q=controller&unscoped_q=controller
https://github.com/decentralized-identity/sidetree/blob/f40d6d9975cd0921c3b0356ee0ee9f6edfdf0c38/lib/core/versions/latest/models/DidPublicKeyModel.ts